src/Nethermind/Nethermind.JsonRpc.Test/Modules/Eth/EthRpcModuleTests.EstimateGas.cs

```diff
@@ -257,4 +258,76 @@ public async Task should_estimate_transaction_with_deployed_code_when_eip3607_en
+        var transaction = JsonSerializer.Deserialize<object>(transactionJson);
+        var stateOverride = JsonSerializer.Deserialize<object>(stateOverrideJson);
+
+        TestSpecProvider specProvider = new(Prague.Instance);
+        using Context ctx = await Context.Create(specProvider);
+
+        string serialized = await ctx.Test.TestEthRpc("eth_estimateGas", transaction, "latest", stateOverride);
+
+        Assert.That(serialized, Is.EqualTo(expectedResult));
+    }
+
+    [TestCase(
+        "When balance and nonce is overriden",
+        """{"from":"0x7f554713be84160fdf0178cc8df86f5aabd33397","to":"0xc200000000000000000000000000000000000000","value":"0x123"}""",
+        """{"0x7f554713be84160fdf0178cc8df86f5aabd33397":{"balance":"0x123", "nonce": "0x123"}}"""
+    )]
+    [TestCase(
+        "When address code is overriden",
+        """{"from":"0x7f554713be84160fdf0178cc8df86f5aabd33397","to":"0xc200000000000000000000000000000000000000","input":"0x60fe47b1112233445566778899001122334455667788990011223344556677889900112233445566778899001122"}""",
+        """{"0xc200000000000000000000000000000000000000":{"code":"0x6080604052348015600e575f80fd5b50600436106030575f3560e01c80632a1afcd914603457806360fe47b114604d575b5f80fd5b603b5f5481565b60405190815260200160405180910390f35b605c6058366004605e565b5f55565b005b5f60208284031215606d575f80fd5b503591905056fea2646970667358221220fd4e5f3894be8e57fc7460afebb5c90d96c3486d79bf47b00c2ed666ab2f82b364736f6c634300081a0033"}}"""
+    )]
+    [TestCase(
+        "When precompile address is changed",
+        """{"from":"0x7f554713be84160fdf0178cc8df86f5aabd33397","to":"0xc200000000000000000000000000000000000000","input":"0xB6E16D27AC5AB427A7F68900AC5559CE272DC6C37C82B3E052246C82244C50E4000000000000000000000000000000000000000000000000000000000000001C7B8B1991EB44757BC688016D27940DF8FB971D7C87F77A6BC4E938E3202C44037E9267B0AEAA82FA765361918F2D8ABD9CDD86E64AA6F2B81D3C4E0B69A7B055"}""",
+        """{"0x0000000000000000000000000000000000000001":{"movePrecompileToAddress":"0xc200000000000000000000000000000000000000", "code": "0x"}}"""
+    )]
+    public async Task Estimate_gas_with_state_override_does_not_affect_other_calls(string name, string transactionJson, string stateOverrideJson)
+    {
+        var transaction = JsonSerializer.Deserialize<object>(transactionJson);
+        var stateOverride = JsonSerializer.Deserialize<object>(stateOverrideJson);
+
+        using Context ctx = await Context.Create();
+
+        var resultOverrideBefore = await ctx.Test.TestEthRpc("eth_estimateGas", transaction, "latest", stateOverride);
+
+        var resultNoOverride = await ctx.Test.TestEthRpc("eth_estimateGas", transaction, "latest");
+
+        var resultOverrideAfter = await ctx.Test.TestEthRpc("eth_estimateGas", transaction, "latest", stateOverride);
+
+        Assert.Multiple(() =>
+        {
+            Assert.That(resultOverrideBefore, Is.EqualTo(resultOverrideAfter), resultOverrideBefore.Replace("\"", "\\\""));
```

Comment:
Prefer JSON over string comparison since it ignores differences like key ordering:

```suggestion
            using FluentAssertions.Json;

            JToken.Parse(resultOverrideBefore).Should().BeEquivalentTo(resultOverrideAfter);
```

Reply:
This looks nice, thanks